src/rsa.rs

```diff
@@ -27,6 +28,30 @@ pub struct RSAPKCS1Padding {
     digestinfo_prefix: &'static [u8],
 }
 
+impl RSAPKCS1Padding {
+    fn pad(&self, msg: &[u8], out: &mut [u8]) -> Result<(), ()> {
+        let digest_len = self.digestinfo_prefix.len() + self.digest_alg.output_len;
+        if out.len() < digest_len + 11 {
+            return Err(());
+        }
+
+        let pad_len = out.len() - digest_len - 3;
+        out[0] = 0;
+        out[1] = 1;
+        for i in 0..pad_len {
+            out[2 + i] = 0xff;
+        }
+        out[2 + pad_len] = 0;
+
+        let mid = 3 + pad_len + self.digestinfo_prefix.len();
+        let (digest_prefix, digest_dst) = out.split_at_mut(mid);
```

Comment:
I suggest you write this as
```rust
let (digest_prefix, digest_value) =
    out[(3 + pad_len)..].split_at_mut(self.digestinfo_prefix.len())];`
```

Then you can avoid the `[pad_len + 3]` below, which reduces the amount of redundancy in the code.